beeai/openshift/deployment-mcp-gateway.yml

```diff
@@ -0,0 +1,78 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mcp-gateway
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: mcp-gateway
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: mcp-gateway
+        deployment: mcp-gateway
+    spec:
+      containers:
+      - args:
+        - mcp_server/gateway.py
+        command:
+        - /usr/bin/python
+        env:
+        - name: SSE_PORT
+          valueFrom:
+            configMapKeyRef:
+              key: SSE_PORT
+              name: mcp-gateway
+
+        - name: GITLAB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              key: GITLAB_TOKEN
+              name: beeai-agent-secrets
+
+        - name: JIRA_TOKEN
+          valueFrom:
+            secretKeyRef:
+              key: JIRA_PERSONAL_TOKEN
+              name: mcp-atlassian-secret
+        - name: JIRA_URL
+          valueFrom:
+            secretKeyRef:
+              key: JIRA_URL
+              name: mcp-atlassian-secret
+
+        image: image-registry.openshift-image-registry.svc:5000/jotnar-prod/mcp-server:prod
+        imagePullPolicy: Always
+        name: mcp-gateway
+        ports:
+        - containerPort: 8000
+          protocol: TCP
+        # TODO: add limits on cpu and memory.
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /git-repos
+          name: mcp-server-git-repos
+        - mountPath: /home/mcp/keytab
```

Comment:
I am not sure how to handle this. Is it possible to mount a secret as a file in OpenShift? If not, what should be the name of this volume? Shall I mount `/home/mcp` and name the volume `mcp-home`? The current path would mount as a directory.

Reply:
The actual path doesn't really matter as long as it is reflected in `KEYTAB_FILE`.